src/query/query_parser/query_parser.rs

```diff
@@ -856,6 +873,20 @@ impl QueryParser {
             ),
         }
     }
+
+    /// Convert a logical AST into a runnable `Query`, taking into account the parser configuration.
+    fn logical_ast_to_query(&self, logical_ast: LogicalAst) -> Box<dyn Query> {
+        match trim_ast(logical_ast) {
```

Comment:
that logic is possibly broken.

We cannot have the check on "is an empty query" done after complex processing like simplification or trim_ast or whatever.

For this feature, an empty query should really be defined off the user input ast.